tests/Support/SupportArrTest.php

```diff
@@ -843,48 +848,59 @@ public function testQuery()
         $this->assertSame('foo=bar&bar=', Arr::query(['foo' => 'bar', 'bar' => '']));
     }
 
-    public function testRandom()
+    #[DataProvider('engines')]
+    public function testRandom($engine)
     {
-        $random = Arr::random(['foo', 'bar', 'baz']);
+        $random = Arr::random(['foo', 'bar', 'baz'], engine: $engine);
         $this->assertContains($random, ['foo', 'bar', 'baz']);
 
-        $random = Arr::random(['foo', 'bar', 'baz'], 0);
+        $random = Arr::random(['foo', 'bar', 'baz'], 0, engine: $engine);
         $this->assertIsArray($random);
         $this->assertCount(0, $random);
 
-        $random = Arr::random(['foo', 'bar', 'baz'], 1);
+        $random = Arr::random(['foo', 'bar', 'baz'], 1, engine: $engine);
         $this->assertIsArray($random);
         $this->assertCount(1, $random);
         $this->assertContains($random[0], ['foo', 'bar', 'baz']);
 
-        $random = Arr::random(['foo', 'bar', 'baz'], 2);
+        $random = Arr::random(['foo', 'bar', 'baz'], 2, engine: $engine);
         $this->assertIsArray($random);
         $this->assertCount(2, $random);
         $this->assertContains($random[0], ['foo', 'bar', 'baz']);
         $this->assertContains($random[1], ['foo', 'bar', 'baz']);
 
-        $random = Arr::random(['foo', 'bar', 'baz'], '0');
+        $random = Arr::random(['foo', 'bar', 'baz'], '0', engine: $engine);
         $this->assertIsArray($random);
         $this->assertCount(0, $random);
 
-        $random = Arr::random(['foo', 'bar', 'baz'], '1');
+        $random = Arr::random(['foo', 'bar', 'baz'], '1', engine: $engine);
         $this->assertIsArray($random);
         $this->assertCount(1, $random);
         $this->assertContains($random[0], ['foo', 'bar', 'baz']);
 
-        $random = Arr::random(['foo', 'bar', 'baz'], '2');
+        $random = Arr::random(['foo', 'bar', 'baz'], '2', engine: $engine);
         $this->assertIsArray($random);
         $this->assertCount(2, $random);
         $this->assertContains($random[0], ['foo', 'bar', 'baz']);
         $this->assertContains($random[1], ['foo', 'bar', 'baz']);
 
         // preserve keys
-        $random = Arr::random(['one' => 'foo', 'two' => 'bar', 'three' => 'baz'], 2, true);
+        $random = Arr::random(['one' => 'foo', 'two' => 'bar', 'three' => 'baz'], 2, true, $engine);
```

Comment:
How about static-styled configuration?

https://github.com/search?q=repo%3Alaravel%2Fframework+%2Fpublic+static+function+%5Cw%2BUsing%5C%28%28%5B%3F%5D%3F%28callable%7CClosure%29%7C%5C%24callback%29%2F&type=code

Reply:
I'm a bit hesitant about that because if state is held in a static context, it could potentially affect other parts of the code that are outside of that specific context, which is undesirable.

At least when it comes to Support\Arr::class, I don't think it should hold state statically.